### PR TITLE
[SMTP] Customize Subject Email (by build template)

### PR DESCRIPTION
**Pull request by michaelact**
Example configuration:
```
spec:
  notification:
    delivery:
      subject: '{{ if eq .Build.Status 3 }}✅ Build Successful ({{ .Build.Substitutions.REPO_NAME }}){{ else }}❌ Build Failure ({{ .Build.Substitutions.REPO_NAME }}){{ end }} | {{.Build.Substitutions._COMMIT_MESSAGE}}'
```

### CODE REVIEW

File: smtp/main_test.go [hunk 1]
```diff
@@ -209,8 +207,30 @@ func TestDefaultEmailTemplate(t *testing.T) {
...
+	if err != nil {
+			t.Fatalf("failed to parse subject template: %v", err)
+	}
+
+	subject := new(bytes.Buffer)
+	if err := subjectTmpl.Execute(subject, view); err != nil {
+			t.Fatalf("failed to execute subject template: %v", err)
+	}
+
+	expectedSubject := "Build some-build-id Status: SUCCESS"
```
Comment (prabenzom) [inline smtp/main_test.go:221]: can we keep the test for when no subject template is set and ensure that passes? then create a new test that tests the subject template?

Comment (michaelact) [reply smtp/main_test.go]: Sure, please kindly check https://github.com/GoogleCloudPlatform/cloud-build-notifiers/pull/201/commits/a217c6264186940042c3627c68dc5039608fe8ac

Comment (michaelact) [reply smtp/main_test.go]: When you have a moment, could we merge it now?

**Review by ddysmhz — APPROVED** [review]

**Review by prabenzom — APPROVED** [review]

**Comment by michaelact** [comment]
Thanks!

**Pull request merged**